executor/src/witgen/jit/block_machine_processor.rs

```diff
@@ -90,15 +92,21 @@ impl<'a, T: FieldElement> BlockMachineProcessor<'a, T> {
             // We iterate over all rows of the block +/- one row.
             -1
         };
-        let identities = (start_row..self.block_size as i32).flat_map(move |row| {
-            self.machine_parts.identities.iter().filter_map(move |&id| {
-                // Filter out identities with next references on the last row.
-                if row as usize == self.block_size - 1 && id.contains_next_ref() {
-                    None
-                } else {
-                    Some((id, row))
-                }
-            })
+        let identities = (start_row..self.block_size as i32).flat_map(|row| {
+            self.machine_parts
+                .identities
+                .iter()
+                .filter_map(|id| {
+                    // Filter out identities with next references on the last row.
+                    if row as usize == self.block_size - 1
+                        && id.contains_next_ref(&intermediate_definitions)
```

Comment:
note: we could filter out identities with next ref once before instead of doing it on each row